superset-frontend/src/datasource/DatasourceEditor.jsx

```diff
@@ -557,6 +557,10 @@ class DatasourceEditor extends React.PureComponent {
     this.setState({ activeTabKey });
   }
 
+  sortMetrics(metrics) {
+    return metrics.sort(({ id: a }, { id: b }) => b - a);
+  }
+
```

Comment:
The sort by id will only work when we change the metric's name because a new id is generated. If we change any other field, the id is preserved, and the sort will fail.

https://user-images.githubusercontent.com/70410625/127668858-32439238-d0a1-442e-97b6-fea56a05ee25.mp4

Reply:
This is actually good news as we don't want that the list is sorted based on the update date but only based on creation. So changing the name will be the only deviation. I will update my comment for the PO to take a decision.